controller/integration/test/controlflow/controlflow_test.go

```diff
@@ -20,90 +20,90 @@ import (
 )
 
 const (
-	objName       = "test-obj"
-	operatorName  = "test-operator"
-	testFinalizer = "operatorkit.giantswarm.io/test-operator"
-	testNamespace = "finalizer-integration-reconciliation-test"
+	objName      = "test-obj"
+	objNamespace = "integration-controlflow-test"
+
+	controllerName = "test-controller"
+	finalizer      = "operatorkit.giantswarm.io/test-controller"
+	resourceName   = "test-resource"
 )
 
 // Test_Finalizer_Integration_Controlflow is an integration test to check that
 // errors during deletion prevent the finalizer from removal.
 func Test_Finalizer_Integration_Controlflow(t *testing.T) {
 	var err error
 
-	var tr *testresource.Resource
+	var resource *testresource.Resource
 	{
-		c := testresource.Config{}
+		c := testresource.Config{
+			Name: resourceName,
+		}
 
-		tr, err = testresource.New(c)
+		resource, err = testresource.New(c)
 		if err != nil {
-			t.Fatal("expected", nil, "got", err)
+			t.Fatalf("err == %v, want %v", err, nil)
 		}
 	}
 
-	// We setup and start the controller.
-	var testWrapper wrapper.Interface
+	var harness wrapper.Interface
 	{
-		c := nodeconfig.Config{
-			Resources: []controller.Resource{
-				tr,
-			},
-
-			Name:      operatorName,
-			Namespace: testNamespace,
-		}
-
-		testWrapper, err = nodeconfig.New(c)
+		harness, err = newHarness(objNamespace, controllerName, resource)
 		if err != nil {
-			t.Fatal("expected", nil, "got", err)
+			t.Fatalf("err == %v, want %v", err, nil)
 		}
+	}
+
+	// Start controller.
+	{
+		controller := harness.Controller()
 
-		testWrapper.MustSetup(testNamespace)
-		defer testWrapper.MustTeardown(testNamespace)
-		newController := testWrapper.Controller()
-		go newController.Boot()
-		<-newController.Booted()
+		go controller.Boot()
+		select {
+		case <-controller.Booted():
+		case <-time.After(30 * time.Second):
+			t.Fatalf("failed to wait for controller to boot")
+		}
 	}
 
-	// We create an object which is valid and wait for the framework to add a
-	// finalizer.
+	// Create an object and wait for the controller to add a finalizer.
+	// Creation is retried because the CRD might still not be ensured.
 	{
 		o := func() error {
 			nodeConfig := &v1alpha1.NodeConfig{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      objName,
-					Namespace: testNamespace,
+					Namespace: objNamespace,
```

Comment:
Maybe I am blind but where is the NS created? It seems to not exist.

Reply:
Your sight is fine. The namespace wasn't created.